alicloud/resource_alicloud_drds_instance.go

```diff
@@ -0,0 +1,174 @@
+		Read:   resourceAliCloudDRDSInstanceRead,
+		Update: resourceAliCloudDRDSInstanceUpdate,
+		Delete: resourceAliCloudDRDSInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateStringLengthInRange(1, 129),
+			},
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validateAllowedStringValue([]string{string(PrivateType), string(PrivateType_),
+					string(PublicType), string(PublicType_)}),
+			},
+			"zone_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"specification": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"pay_type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAllowedStringValue([]string{string(DRDSInstancePostPayType)}),
+			},
+			"vpc_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"vswitch_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"instance_series": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			//`quantity` is similar to `count`, still not sure how to combine these two parameters.
+			"quantity": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"instance_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAliCloudDRDSInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient).drdsconn
+
+	req := drds.CreateCreateDrdsInstanceRequest()
+	req.Description = d.Get("description").(string)
+	req.Type = d.Get("type").(string)
+	req.ZoneId = d.Get("zone_id").(string)
+	req.Specification = d.Get("specification").(string)
+	req.PayType = d.Get("pay_type").(string)
+	req.VpcId = d.Get("vpc_id").(string)
+	req.VswitchId = d.Get("vswitch_id").(string)
+	req.InstanceSeries = d.Get("instance_series").(string)
+	req.Quantity = strconv.Itoa(d.Get("quantity").(int))
+
+	response, err := client.CreateDrdsInstance(req)
+	idList := response.Data.DrdsInstanceIdList.DrdsInstanceId
+
+	if err != nil || len(idList) != 1 {
+		return fmt.Errorf("failed to create DRDS instance with error: %s", err)
+	}
+
+	id := idList[0]
+	d.SetId(id)
+
+	return resourceAliCloudDRDSInstanceUpdate(d, meta)
+}
+
+func resourceAliCloudDRDSInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient).drdsconn
+
+	req := drds.CreateDescribeDrdsInstanceRequest()
+	req.DrdsInstanceId = d.Id()
+
+	res, err := client.DescribeDrdsInstance(req)
+	data := res.Data
+
+	if err != nil || res == nil || data.DrdsInstanceId == "" {
+		return fmt.Errorf("failed to describe DRDS instance with error: %s", err)
+	}
```

Comment:
If there is no instance, Please set id to "" and return nil.

Reply:
Done.